security-analysis/security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysisImpl.java

```diff
@@ -70,69 +73,158 @@ public boolean removeInterceptor(SecurityAnalysisInterceptor interceptor) {
         LoadFlowParameters postContParameters = loadFlowParameters.copy().setVoltageInitMode(LoadFlowParameters.VoltageInitMode.PREVIOUS_VALUES);
 
         network.getVariantManager().allowVariantMultiThreadAccess(true);
+        int contSize = contingenciesProvider.getContingencies(network).size();
+        int count = Math.min(computationManager.getResourcesStatus().getAvailableCores(), contSize);
```

Comment:
I don't see any satisfying way of evaluating the possible number of threads concurrently allocated.
Even taking the machine number of cores is not a satisfying upper limit, since it does not guarantee that the executor service will not run more threads concurrently.

I think the more satisfying solution will be to : 
1. choose the number of concurrent loadflows we want to allo, for instance based on the machine available cores
2. create the corresponding number of variants/workers, and let the the threads created by the executor service access those workers through a pool. The pool will guarantee that workers are not concurrently used.

Reply:
I would propose to go for `Runtime.getRuntime().availableProcessors()` for the number of workers.